bpf/lib/nat.h

```diff
@@ -572,6 +575,12 @@ static __always_inline int snat_v4_rewrite_egress(struct __ctx_buff *ctx,
 	if (csum.offset &&
 	    csum_l4_replace(ctx, off, &csum, 0, sum, flags) < 0)
 		return DROP_CSUM_L4;
+
+#if defined(ENABLE_EGRESS_GATEWAY)
+	if (target && target->egress_gateway)
+		return egress_gw_fib_lookup_and_redirect(ctx, state->to_saddr);
+#endif
+
```

Comment:
Layering-wise, I think we should lift this into `nodeport_snat_fwd_ipv4()`. The `target` should have all the info we need from there (`target.addr == state->to_saddr`, right?)

And `tail_handle_snat_fwd_ipv4()` needs a check to only `send_trace_notify()` for `CTX_ACT_OK`.

Reply:
> Layering-wise, I think we should lift this into `nodeport_snat_fwd_ipv4()`. The `target` should have all the info we need from there (`target.addr == state->to_saddr`, right?)

yep, I was a bit worried about that stray `send_drop_notify_ext` in lib/egress_policies.h :hmmyes:
now drop notifications should get handled by the `tail_handle_snat_fwd_ipv4` caller

> And `tail_handle_snat_fwd_ipv4()` needs a check to only `send_trace_notify()` for `CTX_ACT_OK`.

:bulb: